flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -259,28 +268,28 @@ private void commitUpToCheckpoint(
       long checkpointId)
       throws IOException {
     NavigableMap<Long, byte[]> pendingMap = deltaManifestsMap.headMap(checkpointId, true);
-    List<ManifestFile> manifests = Lists.newArrayList();
```

Comment:
Why is this change needed?
IIRC the `commitUpToCheckpoint` method already handles when the `notifyCheckpointComplete` called with multiple checkpoints.

Reply:
Modifying the previous logic will put multiple checkpoint files into one commit, which is why duplicate data occurs.

Is there something wrong with my understanding?

Reply:
`commitPendingResult` has  a map input `NavigableMap<Long, WriteResult> pendingResults` where the key is the checkpointId, and the value is the merged `WriteResult` for the given checkpoint.

If you check `commitDeltaTxn` you can see that there is a specific case to handle multiple checkpoints:
```
      // To be compatible with iceberg format V2.
      for (Map.Entry<Long, WriteResult> e : pendingResults.entrySet()) {
        // We don't commit the merged result into a single transaction because for the sequential
        // transaction txn1 and txn2, the equality-delete files of txn2 are required to be applied
        // to data files from txn1. Committing the merged one will lead to the incorrect delete
        // semantic.
```

So I think we should be ok if we parametrize the `commitPendingResult` correctly.
This needs to be double checked, but IMHO it should be fine

Reply:
Yes, thanks for your reminder, I checked `commitDeltaTxn`, it works well for multiple checkpoints. This logic does not need to be changed. Rollbacked.